### PR TITLE
Bug 1601257: Part 1 - add docs for scriptworker-scripts.

### PR DESCRIPTION
**Pull request by MihaiTabara**
@rail I stole your stub docs from https://rail.github.io/scriptworkers-gcp/scriptworkers.html and polished them a bit and re-integrated them in the main backbone. Once this is landed and renders nicely in scriptworker-scripts.readthedocs.io/ I'll follow-up with more docs that I have.

### CODE REVIEW

File: docs/scriptworkers-dev.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Dev environment
+
+In case you want to test your changes before pushing to production or
+submitting a PR, you can "force push" your changes to the `dev` branch and it
```
Comment (rail) [inline docs/scriptworkers-dev.md]: You probably also want to mention `dev-$scriptworker` branches.

File: docs/scriptworkers-dockerization.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+Every scriptworker has its own `Dockerfile` in the root directory. The commands
+are limited to the version used in Taskcluster and may not support newer
+features. The file is kept simple and most of the logic is handled by files in
+the `docker.d` directory.
+
+- `docker.d/init.sh`
+
+  This file contains logic that is the same for all workers and supposed to be
+  identical.
```
Comment (rail) [inline docs/scriptworkers-dockerization.md]: They are actually identical and live in the top level directory.

File: docs/scriptworkers-dockerization.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+- `docker.d/init_worker.sh`
```
Comment (rail) [inline docs/scriptworkers-dockerization.md]: This is `$scriptworker/docker.d/init_worker.sh`

File: docs/scriptworkers-dockerization.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  This file contains logic that is worker specific.
+
+Both init files explicitly use shell's `test` to check for all required
+environment variables in order to fail as soon as possible.
+
+- `docker.d/scriptworker.yml` and `docker.d/worker.yml`
```
Comment (rail) [inline docs/scriptworkers-dockerization.md]: `$script/docker.d/worker.yml`

**Review by rail — APPROVED** [review]

**Comment by MihaiTabara** [comment]
@rail adjusted all your comments. 

Just to be clear - this is not the final documentation piece - I merely wanted to commit what I currently have to validate my workflow and the fact that readthedocs can render properly. Once that's in plance, I'll follow-up with some more comprehensive documentation and fix all these nits.

**Pull request merged**